### DISCUSSION TITLE
Completely ignore a target?

### DISCUSSION THREAD

**Discussion by JakeRuss** [ropensci/targets#716] [Help]
Hey Will, 

Is there a way to tell targets to _completely_ ignore a target in a pipeline? Suppose I have pipeline with four targets and I can run different datasets through it. For **Dataset A** all steps apply. **Dataset B** doesn't need the fourth target, can I turn it off completely with a TRUE/FALSE some place?

I've read the docs for `tar_cue` and `tar_cue_skip`, etc.  and I believe they're saying I can skip targets, via setting mode to "never", or some conditionals, but that target will still run if it has never been built. Can I turn that behavior off too?

**Comment by wlandau** [comment]
Other than cues, the closest built-in functionality I can think of is the `names` argument of `tar_make()`.

For your case, it seems like you are programmatically generating sections of the pipeline by `tar_map()`ing over datasets with [static branching](https://books.ropensci.org/targets/static.html). In that case, you could generate the target list and then programmatically remove the targets you don't want by setting them to `NULL`.

```{r}
# _targets.R file
library(targets)
library(tarchetypes)
out <- list(
  a = tar_target(...),
  b = tarchetypes::tar_map(...)
)
out$b$ignore_this <- NULL
out
```

**Reply by JakeRuss** [reply]
In this case, no, I'm not using `tar_map()`. I have a pipeline that performs a generic analysis and I use it for several different clients. Some clients want A, B, and C, others only require A and C (for example). What I want to do is be able to turn B on/off with a parameter or environmental variable. This way there's one pipeline script that I can call from different client projects. 

````
library(targets)

run_B <- FALSE

tar_script({
  list(
    tar_target(data, run_data()),
    tar_target(task_a, run_task_a(data)),
    tar_target(task_b, run_task_b(task_a), cue = tar_cue(run_b, "never")),
    tar_target(task_c, run_task_c(task_a))     
  )
})
````
I realize now I could do the following by using a wrapper function to call tar_make twice, once with A and C always, and then have a toggle for calling tar_make again turning B on and off.

````
library(targets)

tar_script({
  list(
    tar_target(data, run_data()),
    tar_target(task_a, run_task_a(data)),
    tar_target(task_c, run_task_c(task_a))     
  )
})

if(run_b %in% TRUE) {
tar_script({
  list(tar_target(task_b, run_task_b(task_a)
  )
})
}
````
My question was asking if targets had something in-built that ignored a tar_target in the pipeline. Looks like, no, but I think I see an easy work around, so thanks.
